From 1d9228cf62ad7d4eed40f78abafc1829c61c449d Mon Sep 17 00:00:00 2001
From: dianlujitao <dianlujitao@lineageos.org>
Date: Thu, 6 Sep 2018 21:43:14 +0800
Subject: [PATCH 3/6] power: Clean up hint IDs

 * Group MPCTL v3 opcodes according to major type.
 * Remove some unused/dead opcodes.
 * Rename some misleading/inconsistency names.

Change-Id: I3a06930c8fff18a50f77cb230951dbd43d62fed3
---
 performance.h | 149 ++++++++++++++++++++++++--------------------------
 power-660.c   |   2 +-
 power-8952.c  |   2 +-
 power-8953.c  |  12 ++--
 power-8996.c  |   6 +-
 5 files changed, 83 insertions(+), 88 deletions(-)

diff --git a/performance.h b/performance.h
index 23cc099..970d42e 100644
--- a/performance.h
+++ b/performance.h
@@ -47,7 +47,6 @@ enum SCREEN_DISPLAY_TYPE {
 };
 
 enum PWR_CLSP_TYPE {
-    ALL_CPUS_PWR_CLPS_DIS_V3 = 0x40400000, /* v3 resource */
     ALL_CPUS_PWR_CLPS_DIS = 0x101,
 };
 
@@ -105,8 +104,6 @@ enum CPU3_MAX_FREQ_LVL {
 };
 
 enum MIN_CPUS_ONLINE_LVL {
-    CPUS_ONLINE_MIN_BIG = 0x41000000, /* v3 resource */
-    CPUS_ONLINE_MIN_LITTLE = 0x41000100, /* v3 resource */
     CPUS_ONLINE_MIN_2 = 0x702,
     CPUS_ONLINE_MIN_3 = 0x703,
     CPUS_ONLINE_MIN_4 = 0x704,
@@ -115,8 +112,6 @@ enum MIN_CPUS_ONLINE_LVL {
 };
 
 enum MAX_CPUS_ONLINE_LVL {
-    CPUS_ONLINE_MAX_LIMIT_BIG = 0x41004000, /* v3 resource */
-    CPUS_ONLINE_MAX_LIMIT_LITTLE = 0x41004100, /* v3 resource */
     CPUS_ONLINE_MAX_LIMIT_1 = 0x8FE,
     CPUS_ONLINE_MAX_LIMIT_2 = 0x8FD,
     CPUS_ONLINE_MAX_LIMIT_3 = 0x8FC,
@@ -220,7 +215,6 @@ enum SCREEN_PWR_CLPS_LVL {
 
 enum THREAD_MIGRATION_LVL {
     THREAD_MIGRATION_SYNC_OFF = 0x1400,
-    THREAD_MIGRATION_SYNC_ON_V3 = 0x4241C000
 };
 
 enum INTERACTIVE_IO_BUSY_LVL {
@@ -229,8 +223,6 @@ enum INTERACTIVE_IO_BUSY_LVL {
 };
 
 enum SCHED_BOOST_LVL {
-    SCHED_BOOST_ON_V3 = 0x40C00000, /* v3 resource */
-    SCHED_BOOST_ON_V3_LVL = 0x40C02000, /* v3 resource */
     SCHED_BOOST_ON = 0x1E01,
 };
 
@@ -271,7 +263,6 @@ enum CPU7_MAX_FREQ_LVL {
 };
 
 enum SCHED_PREFER_IDLE {
-    SCHED_PREFER_IDLE_DIS_V3 = 0x40C04000,
     SCHED_PREFER_IDLE_DIS = 0x3E01,
 };
 
@@ -282,78 +273,82 @@ enum SCHED_MIGRATE_COST_CHNG {
 /**
  * MPCTL v3 opcodes
  */
-enum MAX_FREQ_CLUSTER_BIG {
-    MAX_FREQ_BIG_CORE_0         = 0x40804000,
-    MAX_FREQ_BIG_CORE_0_LVL   = 0x40806000,
-};
-
-enum MAX_FREQ_CLUSTER_LITTLE {
-    MAX_FREQ_LITTLE_CORE_0      = 0x40804100,
-    MAX_FREQ_LITTLE_CORE_0_LVL  = 0x40806100,
-};
-
-enum MIN_FREQ_CLUSTER_BIG {
-    MIN_FREQ_BIG_CORE_0         = 0x40800000,
-    MIN_FREQ_BIG_CORE_0_LVL     = 0x40802000,
-};
-
-enum MIN_FREQ_CLUSTER_LITTLE {
-    MIN_FREQ_LITTLE_CORE_0      = 0x40800100,
-    MIN_FREQ_LITTLE_CORE_0_LVL  = 0x40802100,
-};
-
-enum INTERACTIVE_CLUSTER_BIG {
-    ABOVE_HISPEED_DELAY_BIG     = 0x41400000,
-    ABOVE_HISPEED_DELAY_BIG_LVL = 0x41402000,
-    GO_HISPEED_LOAD_BIG         = 0x41410000,
-    HISPEED_FREQ_BIG            = 0x41414000,
-    IO_IS_BUSY_BIG              = 0x4141A000,
-    TARGET_LOADS_BIG            = 0x41420000,
-    IGNORE_HISPEED_NOTIF_BIG    = 0x41438000,
-};
-
-enum INTERACTIVE_CLUSTER_LITTLE {
-    ABOVE_HISPEED_DELAY_LITTLE  = 0x41400100,
-    GO_HISPEED_LOAD_LITTLE      = 0x41410100,
-    HISPEED_FREQ_LITTLE         = 0x41414100,
-    TARGET_LOADS_LITTLE         = 0x41420100,
-    IGNORE_HISPEED_NOTIF_LITTLE = 0x41438100,
-};
-
+/* 0x1 */
+enum POWER_COLLAPSE {
+    ALL_CPUS_PWR_CLPS_DIS_V3 = 0x40400000,
+};
+
+/* 0x2 */
+enum CPUFREQ {
+    MIN_FREQ_BIG_CORE_0                 = 0x40800000,
+    MIN_FREQ_BIG_CORE_0_RESIDX          = 0x40802000,
+    MIN_FREQ_LITTLE_CORE_0              = 0x40800100,
+    MIN_FREQ_LITTLE_CORE_0_RESIDX       = 0x40802100,
+    MAX_FREQ_BIG_CORE_0                 = 0x40804000,
+    MAX_FREQ_BIG_CORE_0_RESIDX          = 0x40806000,
+    MAX_FREQ_LITTLE_CORE_0              = 0x40804100,
+    MAX_FREQ_LITTLE_CORE_0_RESIDX       = 0x40806100,
+};
+
+/* 0x3 */
+enum SCHED {
+    SCHED_BOOST_ON_V3                   = 0x40C00000,
+    SCHED_PREFER_IDLE_DIS_V3            = 0x40C04000,
+    SCHED_MIGRATE_COST_SET_V3           = 0x40C08000,
+    SCHED_SMALL_TASK                    = 0x40C0C000,
+    SCHED_MOSTLY_IDLE_LOAD              = 0x40C10000,
+    SCHED_MOSTLY_IDLE_NR_RUN            = 0x40C14000,
+    SCHED_GROUP_ON                      = 0x40C28000,
+    SCHED_SPILL_NR_RUN                  = 0x40C2C000,
+    SCHED_RESTRICT_CLUSTER_SPILL        = 0x40C34000,
+    SCHED_GROUP_UP_MIGRATE              = 0x40C54000,
+    SCHED_GROUP_DOWN_MIGRATE            = 0x40C58000,
+};
+
+/* 0x4 */
+enum CORE_HOTPLUG {
+    CPUS_ONLINE_MIN_BIG                 = 0x41000000,
+    CPUS_ONLINE_MAX_BIG                 = 0x41004000,
+    CPUS_ONLINE_MIN_LITTLE              = 0x41000100,
+    CPUS_ONLINE_MAX_LITTLE              = 0x41004100,
+};
+
+/* 0x5 */
+enum INTERACTIVE {
+    ABOVE_HISPEED_DELAY_BIG             = 0x41400000,
+    ABOVE_HISPEED_DELAY_BIG_RESIDX      = 0x41402000,
+    GO_HISPEED_LOAD_BIG                 = 0x41410000,
+    HISPEED_FREQ_BIG                    = 0x41414000,
+    TARGET_LOADS_BIG                    = 0x41420000,
+    IGNORE_HISPEED_NOTIF_BIG            = 0x41438000,
+    ABOVE_HISPEED_DELAY_LITTLE          = 0x41400100,
+    ABOVE_HISPEED_DELAY_LITTLE_RESIDX   = 0x41402100,
+    GO_HISPEED_LOAD_LITTLE              = 0x41410100,
+    HISPEED_FREQ_LITTLE                 = 0x41414100,
+    TARGET_LOADS_LITTLE                 = 0x41420100,
+    IGNORE_HISPEED_NOTIF_LITTLE         = 0x41438100,
+};
+
+/* 0x6 */
 enum CPUBW_HWMON {
-    CPUBW_HWMON_MIN_FREQ        = 0x41800000,
-    CPUBW_HWMON_MIN_FREQ_LVL    = 0x41802000,
-    CPUBW_HWMON_V1              = 0x4180C000,
-    LOW_POWER_CEIL_MBPS         = 0x41810000,
-    LOW_POWER_IO_PERCENT        = 0x41814000,
-    CPUBW_HWMON_SAMPLE_MS       = 0x41820000,
-};
-
-enum SCHEDULER {
-    SCHED_SMALL_TASK_DIS        = 0x40C0C000,
-    SCHED_IDLE_LOAD_DIS         = 0x40C10000,
-    SCHED_IDLE_NR_RUN_DIS       = 0x40C14000,
-    SCHED_IDLE_NR_RUN           = 0x40C2C000,
-    SCHED_IDLE_NR_RUN_LVL       = 0x40C2E000,
-    SCHED_IDLE_RESTRICT_CLUSTER = 0x40C34000,
-    SCHED_IDLE_RESTRICT_CLUSTER_LVL = 0x40C36000,
-    SCHED_GROUP_ON              = 0x40C28000,
-    SCHED_GROUP_UP_MIGRATE      = 0x40C54000,
-    SCHED_GROUP_DOWN_MIGRATE    = 0x40C58000,
-    SCHED_FREQ_AGGR_THRH_LVL    = 0x40C5E000,
-};
-
-enum STORAGE {
-    STOR_CLK_SCALE_DIS          = 0x42C10000,
-    STOR_CLK_SCALE_DIS_LVL      = 0x42C12000,
+    CPUBW_HWMON_MIN_FREQ                = 0x41800000,
+    CPUBW_HWMON_MIN_FREQ_RESIDX         = 0x41802000,
+    CPUBW_HWMON_HYST_OPT                = 0x4180C000,
+    LOW_POWER_CEIL_MBPS                 = 0x41810000,
+    LOW_POWER_IO_PERCENT                = 0x41814000,
+    CPUBW_HWMON_SAMPLE_MS               = 0x41820000,
 };
 
+/* 0xA */
 enum GPU {
-    GPU_MIN_PWRLVL_BOOST        = 0x42804000,
-    GPU_MIN_FREQ                = 0x4280C000,
-    GPU_MIN_FREQ_LVL            = 0x4280E000,
-    GPU_MAX_FREQ                = 0x42810000,
-    GPUBW_MIN_FREQ              = 0x42814000,
+    GPU_MIN_POWER_LEVEL                 = 0x42804000,
+    GPU_MAX_POWER_LEVEL                 = 0x42808000,
+    GPU_MIN_FREQ                        = 0x4280C000,
+    GPU_MIN_FREQ_RESIDX                 = 0x4280E000,
+    GPU_MAX_FREQ                        = 0x42810000,
+    GPU_MAX_FREQ_RESIDX                 = 0x42812000,
+    GPUBW_MIN_FREQ                      = 0x42814000,
+    GPUBW_MAX_FREQ                      = 0x42818000,
 };
 
 #ifdef __cplusplus
diff --git a/power-660.c b/power-660.c
index f9d41d8..0a9ef37 100644
--- a/power-660.c
+++ b/power-660.c
@@ -122,7 +122,7 @@ static void process_video_encode_hint(void *metadata)
                     GO_HISPEED_LOAD_BIG, 0x5F,
                     ABOVE_HISPEED_DELAY_BIG, 0x4,
                     TARGET_LOADS_BIG, 0x5F,
-                    SCHED_IDLE_NR_RUN, 0X5,
+                    SCHED_SPILL_NR_RUN, 0X5,
                     CPUBW_HWMON_SAMPLE_MS, 0xA
                 };
                 memcpy(resource_values, res, MIN_VAL(sizeof(resource_values), sizeof(res)));
diff --git a/power-8952.c b/power-8952.c
index e72526b..c0f3a67 100644
--- a/power-8952.c
+++ b/power-8952.c
@@ -61,7 +61,7 @@ static int profile_high_performance[] = {
 };
 
 static int profile_power_save[] = {
-    CPUS_ONLINE_MAX_LIMIT_BIG, 0x0,
+    CPUS_ONLINE_MAX_BIG, 0x0,
     CPU0_MAX_FREQ_NONTURBO_MAX,
     CPU4_MAX_FREQ_NONTURBO_MAX,
 };
diff --git a/power-8953.c b/power-8953.c
index d272a23..78ee3df 100644
--- a/power-8953.c
+++ b/power-8953.c
@@ -60,15 +60,15 @@ static int profile_high_performance[] = {
     CPUS_ONLINE_MIN_BIG, 0x4,
     MIN_FREQ_BIG_CORE_0, 0xFFF,
     MIN_FREQ_LITTLE_CORE_0, 0xFFF,
-    GPU_MIN_PWRLVL_BOOST, 0x1,
+    GPU_MIN_POWER_LEVEL, 0x1,
     SCHED_PREFER_IDLE_DIS_V3, 0x1,
-    SCHED_SMALL_TASK_DIS, 0x1,
-    SCHED_IDLE_NR_RUN_DIS, 0x1,
-    SCHED_IDLE_LOAD_DIS, 0x1,
+    SCHED_SMALL_TASK, 0x1,
+    SCHED_MOSTLY_IDLE_NR_RUN, 0x1,
+    SCHED_MOSTLY_IDLE_LOAD, 0x1,
 };
 
 static int profile_power_save[] = {
-    CPUS_ONLINE_MAX_LIMIT_BIG, 0x1,
+    CPUS_ONLINE_MAX_BIG, 0x1,
     MAX_FREQ_BIG_CORE_0, 0x3bf,
     MAX_FREQ_LITTLE_CORE_0, 0x300,
 };
@@ -79,7 +79,7 @@ static int profile_bias_power[] = {
 };
 
 static int profile_bias_performance[] = {
-    CPUS_ONLINE_MAX_LIMIT_BIG, 0x4,
+    CPUS_ONLINE_MAX_BIG, 0x4,
     MIN_FREQ_BIG_CORE_0, 0x540,
 };
 
diff --git a/power-8996.c b/power-8996.c
index 4ad56ea..f782299 100644
--- a/power-8996.c
+++ b/power-8996.c
@@ -64,7 +64,7 @@ static int profile_high_performance[] = {
 };
 
 static int profile_power_save[] = {
-    CPUS_ONLINE_MAX_LIMIT_BIG, 0x1,
+    CPUS_ONLINE_MAX_BIG, 0x1,
     MAX_FREQ_BIG_CORE_0, 0x3E8,
     MAX_FREQ_LITTLE_CORE_0, 0x3E8,
 };
@@ -75,8 +75,8 @@ static int profile_bias_power[] = {
 };
 
 static int profile_bias_performance[] = {
-    CPUS_ONLINE_MAX_LIMIT_BIG, 0x2,
-    CPUS_ONLINE_MAX_LIMIT_LITTLE, 0x2,
+    CPUS_ONLINE_MAX_BIG, 0x2,
+    CPUS_ONLINE_MAX_LITTLE, 0x2,
     MIN_FREQ_BIG_CORE_0, 0x578,
 };
 
-- 
2.17.1

